docs/asb/README.md

```diff
@@ -0,0 +1,129 @@
+# XMR to BTC Atomic Swap - Automated Swap Backend (ASB)
+
+## Quick Start ASB
+
+1. Ensure that you have the Monero Wallet RPC running with `--wallet-dir` and `--disable-rpc-login`:
+   1. `monero-wallet-rpc --stagenet --daemon-host STAGENET-NODE-URL --rpc-bind-port STAGENET-NODE-PORT --disable-rpc-login --wallet-dir PATH/TO/WALLET/DIR`
+2. Run the ASB in terminal: `./asb start`
+3. Follow the setup wizard in the terminal
+
+Public stagenet nodes:
+
+- `monero-stagenet.exan.tech:38081`
+- `stagenet.community.xmr.to:38081`
+
+Run `./asb --help` for more information.
+
+## ASB Details
+
+The ASB is a long running daemon that acts as the trading partner to the swap CLI.
+The CLI user is buying XMR (i.e. receives XMR, sends BTC), the ASB service provider is selling XMR (i.e. sends XMR, receives BTC).
+The ASB can handle multiple swaps with different peers concurrently.
+The ASB communicates with the CLI on various [libp2p](https://libp2p.io/)-based network protocols.
+
+Both the ASB and the CLI can be run by anybody.
+The CLI is designed to run one specific swap against an ASB.
+The ASB is designed to run 24/7 as a daemon that responds to CLIs connecting.
+Since the ASB is a long running task we specify the person running an ASB as service provider.
+
+### ASB discovery
+
+Currently, there is no automated discovery for service providers running an ASB.
+A service provider has to manually provide the connection details to users that will run the CLI.
+
+[Libp2p addressing](https://docs.libp2p.io/concepts/addressing/) is used to identify a service provider by multi-address and peer-id.
+The Peer-ID is printed upon startup of the ASB.
+The multi-address typically consists of IP-address or URL (if DNS entry configured) of the service provider.
+
+Each service provider running an ASB can decide how/where to share these connection details.
+
+![Service Provider scenarios](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/comit-network/xmr-btc-swap/d2cf45d8b9f0c2e180cd85aa034f370965adc11c/docs/asb/diagrams/cli-asb-overview.puml)
+
+Eventually, more elaborate discovery mechanisms can be added.
+
+The **CLI** user can specify a service providers's multiaddress and peer-id with `--seller-addr` and `--seller-peer`, see `./swap --help` for details.
+
+### Setup Details
+
+In order to understand the different components of the ASB and CLI better here is a component diagram showcasing the ASB and CLI setup using public Bitcoin and Monero infrastructure:
+
+![Service Provider scenarios](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/comit-network/xmr-btc-swap/363ce1cdf6fe6478736ff91e1458d650c2319248/docs/asb/diagrams/cli-asb-components-asb-pub-nodes.puml)
+
+Contrary, here is a diagram that showcases an ASB running it's own blockchain infrastructure:
```

Comment:
Minor nit: The ASB is never really running it's own infrastructure, it is just an application. The service provider is running the infrastructure.